ethereum/contracts/Migrations.sol

```diff
@@ -0,0 +1,26 @@
+pragma solidity >=0.4.22 <0.7.0;
+
+contract Migrations {
+  address public owner;
+
+  // A function with the signature `last_completed_migration()`, returning a uint, is required.
+  uint public last_completed_migration;
+
+  modifier restricted() {
+    if (msg.sender == owner) _;
+  }
+
+  constructor() {
+    owner = msg.sender;
+  }
+
+  // A function with the signature `setCompleted(uint)` is required.
+  function setCompleted(uint completed) restricted {
```

Comment:
So this marks the smart contract completed? And another file deploys the contract  ?

Reply:
I honestly don't know much about this contract other than it keeps track of migrations. It was auto-generated when I ran `truffle init` in the repo.